In [1]:
import numpy as np
import pandas as pd
import scipy
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
df = pd.read_csv('epi_r.csv')

In [3]:
df.head()

title  rating  calories  protein  \
0              Lentil, Apple, and Turkey Wrap    2.500     426.0     30.0   
1  Boudin Blanc Terrine with Red Onion Confit    4.375     403.0     18.0   
2                Potato and Fennel Soup Hodge    3.750     165.0      6.0   
3             Mahi-Mahi in Tomato Olive Sauce    5.000       NaN      NaN   
4                    Spinach Noodle Casserole    3.125     547.0     20.0   

    fat  sodium  #cakeweek  #wasteless  22-minute meals  3-ingredient recipes  \
0   7.0   559.0        0.0         0.0              0.0                   0.0   
1  23.0  1439.0        0.0         0.0              0.0                   0.0   
2   7.0   165.0        0.0         0.0              0.0                   0.0   
3   NaN     NaN        0.0         0.0              0.0                   0.0   
4  32.0   452.0        0.0         0.0              0.0                   0.0   

    ...    yellow squash  yogurt  yonkers  yuca  zucchini  cookbooks  \
0   ...              0.0     0.0      0.0   0.0       0.0        0.0   
1   ...              0.0     0.0      0.0   0.0       0.0        0.0   
2   ...              0.0     0.0      0.0   0.0       0.0        0.0   
3   ...              0.0     0.0      0.0   0.0       0.0        0.0   
4   ...              0.0     0.0      0.0   0.0       0.0        0.0   

   leftovers  snack  snack week  turkey  
0        0.0    0.0         0.0     1.0  
1        0.0    0.0         0.0     0.0  
2        0.0    0.0         0.0     0.0  
3        0.0    0.0         0.0     0.0  
4        0.0    0.0         0.0     0.0  

[5 rows x 680 columns]

In [15]:
null_count = df.isnull().sum()

In [17]:
null_count[null_count>0]

calories    4117
protein     4162
fat         4183
sodium      4119
dtype: int64

In [18]:
recipes = df.dropna()

In [19]:
recipes.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 15864 entries, 0 to 20051
Columns: 680 entries, title to turkey
dtypes: float64(679), object(1)
memory usage: 82.4+ MB


Removed the null values in the dietary information column. Will run to see how it performs

In [31]:
from sklearn.svm import SVR
svr = SVR()
X = recipes.drop(['rating','title'], 1)
Y = recipes['rating']

In [32]:
svr.fit(X,Y)

SVR(C=1.0, cache_size=200, coef0=0.0, degree=3, epsilon=0.1, gamma='auto',
  kernel='rbf', max_iter=-1, shrinking=True, tol=0.001, verbose=False)

In [33]:
svr.score(X, Y)

0.20120576501871634

Pretty good improvement, from 3.8% to 20%. The other two ideas are to clean up the data further and possibly transform the target variable into binary, to make it a classification problem.